cluster-autoscaler/simulator/drainability/rules/pdb/rule.go

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pdb
+
+import (
+	"fmt"
+
+	apiv1 "k8s.io/api/core/v1"
+	"k8s.io/autoscaler/cluster-autoscaler/simulator/drainability"
+	"k8s.io/autoscaler/cluster-autoscaler/utils/drain"
+)
+
+// Rule is a drainability rule on how to handle pods with pdbs.
+type Rule struct{}
+
+// New creates a new Rule.
+func New() *Rule {
+	return &Rule{}
+}
+
+// Drainable decides how to handle pods with pdbs on node drain.
+func (Rule) Drainable(drainCtx *drainability.DrainContext, pod *apiv1.Pod) drainability.Status {
+	for _, pdb := range drainCtx.RemainingPdbTracker.MatchingPdbs(pod) {
+		if pdb.Status.DisruptionsAllowed < 1 {
```

Comment:
nit: Why not just rely on `CanRemovePods` here?

Reply:
CanRemovePods takes all pods. That would be a worse runtime. Also, the `inParallel` return value is not currently being used in this context, so MatchingPdbs(pod) is a better pattern for the time being.

If we want to do multi-pod matching with pdb rules, we would need to rethink how to implement this with drainability rules, since a single rule handles a single pod. Likely the pdb tracker would have to cache the multi-pod result.